src/main/java/seedu/address/storage/delivery/JsonAdaptedDelivery.java

```diff
@@ -82,7 +91,20 @@ public Delivery toModelType() throws IllegalValueException {
         }
         final Order modelOrder = new Order(order);
 
-        return new Delivery(modelName, modelPhone, modelAddress, modelOrder);
+        if (endTime == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Time.class.getSimpleName()));
+        }
+
+        /*
+        try {
+            LocalDateTime.parse(endTime, DateTimeFormatter.ofPattern("dd MMMM yyyy HH:mm:ss"));
+        } catch (DateTimeParseException e) {
+            throw new IllegalValueException(Time.MESSAGE_CONSTRAINTS);
+        }*/
```

Comment:
For the other parts of the code, this section here is to check if the JSON info is valid (ie. is the data file corrupt). However, this does not work for this implementation of Time, as we are not storing user input, but we are storing the endTime. Hence we cannot pass the endTime field through the same isValidQuantity/isValidName/etc. method, as that regex is used to test user input not to test this.

I removed this because I am handling the parsing inside the Time class

Reply:
Would it be better to just delete it

Reply:
ok true, fixed in latest commit